src/transformers/models/blt/modular_blt.py

```diff
@@ -0,0 +1,976 @@
+            dropout=0.0 if not self.training else self.dropout,
+            scaling=self.scaling,
+            **kwargs,
+        )
+        attn_output = attn_output.reshape(bsz, q_len, -1).contiguous()
+        attn_output = self.o_proj(attn_output)
+        attn_output = attn_output + hidden_states
+        return attn_output, attn_weights
+
+
+class BltGlobalTransformer(nn.Module):
+    def __init__(self, config: BltGlobalTransformerConfig):
+        super().__init__()
+        self.config = config
+        self.layers = nn.ModuleList()
+        for layer_idx in range(config.num_hidden_layers):
+            self.layers.append(BltTransformerLayer(config, layer_idx))
+        self.rotary_emb = BltRotaryEmbedding(config=config)
+
+    def forward(
+        self,
+        input_embeds: torch.Tensor,
+        attention_mask: Optional[torch.Tensor] = None,
+        position_ids: Optional[torch.LongTensor] = None,
+        past_key_values: Optional[Cache] = None,
+        cache_position: Optional[torch.LongTensor] = None,
+        **kwargs: Unpack[TransformersKwargs],
+    ):
+        batch_size, seq_len, _ = input_embeds.shape
+        hidden_states = input_embeds
+        hidden_states = F.dropout(hidden_states, p=self.config.dropout, training=self.training)
+        if position_ids is None:
+            position_ids = (
+                torch.arange(input_embeds.shape[1], device=input_embeds.device).unsqueeze(0).expand(batch_size, -1)
+            )
+        position_embeddings = self.rotary_emb(hidden_states, position_ids)
+        for i, layer in enumerate(self.layers):
+            hidden_states = layer(
+                hidden_states,
+                position_embeddings=position_embeddings,
+                attention_mask=attention_mask,
+                past_key_values=past_key_values,
+                cache_position=cache_position,
+                **kwargs,
+            )
+        return hidden_states
+
+
+@auto_docstring
+class BltPreTrainedModel(MllamaPreTrainedModel):
+    config: BltConfig
+    base_model_prefix = "model"
+    supports_gradient_checkpointing = True
+    _no_split_modules = ["BltTransformerLayer", "BltLocalEncoder", "BltLocalDecoder", "BltGlobalTransformer"]
+
+    _supports_static_cache = False  # static cache cannot have different shapes for each layer
+    _supports_sdpa = True
+    _supports_flash_attn = True
+    _supports_flex_attn = True
+    _supports_attention_backend = True
+    _can_record_outputs = {
+        "hidden_states": OutputRecorder(BltTransformerLayer, index=0, layer_name="local_decoder"),
+        "attentions": OutputRecorder(BltSelfAttention, index=1, layer_name="local_decoder"),
+        "encoder_attentions": OutputRecorder(BltSelfAttention, index=1, layer_name="local_encoder"),
+        "global_attentions": OutputRecorder(BltSelfAttention, index=1, layer_name="global_transformer"),
+    }
+
+    def _init_weights(self, module):
+        PreTrainedModel._init_weights(self, module)
```

Comment:
If we don't want to inherit the one from Mllama (because there is no need to have it, as it's the same as the PreTrainedModel one anyway), let's do:
```suggestion
    def _init_weights(self, module):
        raise AttributeError("No need to inherit it!")
```
which will make modular skip it